diff --git a/dist/index.es.js b/dist/index.es.js
index 051b5c4c3b5a857083cefe7758c466d4ee4f7038..b8513350f540c9a5aa14b1c7242e6e11738c466a 100644
--- a/dist/index.es.js
+++ b/dist/index.es.js
@@ -130,6 +130,7 @@ class TauriPty {
                 for (;;) {
                     const data = yield invoke('plugin:pty|read', { pid: this.pid });
                     this._onData.fire(data);
+                    yield new Promise(r => setTimeout(r, 0));
                 }
             }
             catch (e) {
